checks/kubernetes_resource.go

```diff
@@ -95,7 +99,6 @@ func (c *KubernetesResourceChecker) Check(ctx *context.Context, check v1.Kuberne
 
 	for i := range check.Resources {
 		resource := check.Resources[i]
-
 		resource.SetAnnotations(map[string]string{annotationkey: ctx.Canary.ID()})
```

Comment:
We need to use check id vs canary-id as there could be more than 1 check per canary

I think a label might be better - we can then just track the kinds, do a single `kubectl delete {kind} --all-namespaces -l ` - If you remove a kind a kind from an existing check then it gets orphaned.